mlrun/runtimes/pod.py

```diff
@@ -1347,20 +1347,26 @@ def _resolve_build_with_mlrun(self, with_mlrun: typing.Optional[bool] = None):
 
     def _build_image(
         self,
-        builder_env,
-        force_build,
-        mlrun_version_specifier,
-        show_on_failure,
-        skip_deployed,
-        watch,
-        is_kfp,
-        with_mlrun,
+        builder_env: dict,
+        force_build: bool,
+        mlrun_version_specifier: typing.Optional[bool],
+        show_on_failure: bool,
+        skip_deployed: bool,
+        watch: bool,
+        is_kfp: bool,
+        with_mlrun: typing.Optional[bool],
     ):
         # When we're in pipelines context we must watch otherwise the pipelines pod will exit before the operation
         # is actually done. (when a pipelines pod exits, the pipeline step marked as done)
         if is_kfp:
             watch = True
 
+        if skip_deployed and self.requires_build() and not self.is_deployed():
+            logger.warning(
+                f"Although {skip_deployed=}, build may be triggered due to function configuration. "
```

Comment:
f"Even though {skip_deployed=}, the build might be triggered due to the function's configuration. "